src/core/MIGRATION.md

```diff
@@ -621,7 +750,20 @@ Concerns around ownership or duplication of a given module should be raised and
 
 A great outcome is a module being deleted altogether because it isn't used or it was used so lightly that it was easy to refactor away.
 
-### Provide plugin extension points decoupled from angular.js
+If it is determined that your plugin is going to own any UI modules that other plugins depend on, you'll want to migrate these quickly so that there's time for downstream plugins to update their imports. This will ultimately involve moving the module code into your plugin, and exposing it via your setup/start contracts, or as static code from your `plugin/index.ts`.
```

Comment:
I think in this case, we should recommend a plugin developer to consult with us on where is the correct location to put something.
Otherwise we might end up with plugins owning generic modules, while we could recommend the correct place to migrate them into (probably places like `kibana_react` and `kibana_utils`, but I'm not sure we want to mention them here?).

Reply:
Yeah, I added the qualifier "If it is determined that your plugin is going to own any UI modules that other plugins depend on" because I didn't want to go too far down the road of what-should-live-where. What I'm trying to communicate here is to prioritize items that will be needed by other plugins, and export them via your `legacy.ts` shim or your `index.ts`.

Either way I will update this to be more explicit about asking for help